### PR TITLE
Fix typo in github link

### LINKED ISSUE #1235: Update link to `wordpress.github.io/openverse` docs on API documentation site
## Problem

The API documentation site still links to `wordpress.githb.io/openverse` (without the `u`) and should point to `docs.openverse.org` instead.

https://github.com/WordPress/openverse/blob/697f62f01a32cb7fcf2f4a7627650a113cba40da/api/catalog/api/docs/README.md#L3

## Additional context

Marked as `high` becuase `githb.io` is a parked domain page and an error of that kind on the frontpage of our API documentation is not a good representation of the project.

### PR DESCRIPTION
**Pull request by capitan-beto**
## Fixes

Fixes #1235 by @sarayourfriend 

## Description

Fix typo in github link of the /api/catalog/api/docs README. Now the link redirects to the correct URL (while previous redirects 
to another domain).

Captura desde 2023-04-17 18-26-08

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

### CODE REVIEW

File: api/catalog/api/docs/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 This documentation is focused towards consumers who are using the Openverse API.
 To contribute to the development of the Openverse API, please refer to the
-[Openverse developer documentation](https://wordpress.githb.io/openverse/).
+[Openverse developer documentation](https://wordpress.github.io/openverse/).
```
Comment (AetherUnbound) [inline api/catalog/api/docs/README.md:3]: It looks like this directs to our new documentation site, https://docs.openverse.org/. Per the original issue, I believe we should update this to that instead:
```suggestion
[Openverse developer documentation](https://docs.openverse.org/).
```

**Review by AetherUnbound — DISMISSED** [review]
Thanks so much for your contribution @capitan-beto! I have one suggested change, given that the updated URL redirects to a different page.

File: api/catalog/api/docs/README.md (hunk 1 above)
Comment (capitan-beto) [reply api/catalog/api/docs/README.md]: Hi! thanks for the feedback, i've already corrected and pushed the issue. 

Captura desde 2023-04-17 18-58-41

**Review by krysal — APPROVED** [review]

**Review by zackkrida — APPROVED** [review]

**Comment by zackkrida** [comment]
Thank you @capitan-beto and congratulations on your first Openverse contribution 🥳

**Review by AetherUnbound — APPROVED** [review]
Excellent!

**Pull request merged**